File: Projects/Project_Cinema Website/backend/src/main/java/com/cinema_website/backend/service/ReviewService.java
package com.cinema_website.backend.service;

import com.cinema_website.backend.dto.ReviewDTO;
import com.cinema_website.backend.repository.ReviewRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class ReviewService {

    private final ReviewRepository reviewRepository;

    @Autowired
    public ReviewService(ReviewRepository reviewRepository) {
        this.reviewRepository = reviewRepository;
    }

    // GET
    public ReviewDTO getReviewById(int reviewId) {
        return reviewRepository.getReviewById(reviewId);
    }

    public List<ReviewDTO> getAllReviews() {
        return reviewRepository.getAllReviews();
    }

    public List<ReviewDTO> getReviewsByTarget(int targetId, String targetTable) {
        return reviewRepository.getReviewsByTarget(targetId, targetTable);
    }
    public List<ReviewDTO> getReviewsByUser(int userId) {
        return reviewRepository.getReviewsByUser(userId);
    }

    // INSERT
    public boolean addReview(ReviewDTO reviewDTO) {
        return reviewRepository.addReview(reviewDTO);
    }

    // UPDATE
    public boolean updateReviewById(int reviewId, ReviewDTO updatedReview) {
        return reviewRepository.updateReviewById(reviewId, updatedReview);
    }

    // DELETE
    public boolean deleteReviewById(int reviewId) {
        return reviewRepository.deleteReviewById(reviewId);
    }
}

File: Projects/Project_Cinema Website/frontend/src/services/ReviewService.js
import axios from "axios";

const baseURL = 'http://localhost:8080/api/reviews'; // Replace with your actual API endpoint

class ReviewService {

    // GET
    static async getReviewById(reviewId) {
        const response = await axios.get(`${baseURL}/getReview/${reviewId}`);
        return response.data;
    }

    static async getAllReviews() {
        const response = await axios.get(`${baseURL}/getAllReviews`);
        return response.data;
    }

    static async getReviewsByTarget(targetId, targetTable) {
        const response = await axios.get(`${baseURL}/getReviewsByTarget/${targetId}/${targetTable}`);
        return response.data;
    }

    static async getReviewsByUser(userId) {
        const response = await axios.get(`${baseURL}/getReviewsByUser/${userId}`);
        return response.data;
    }



    // POST
    static async addReview(newReview) {
        const response = await axios.post(`${baseURL}/addReview`, newReview);
        return response.data;
    }

    // PUT (Update)
    static async updateReviewById(reviewId, updatedReview) {
        const response = await axios.put(`${baseURL}/updateReview/${reviewId}`, updatedReview);
        return response.data;
    }

    // DELETE
    static async deleteReviewById(reviewId) {
        const response = await axios.delete(`${baseURL}/deleteReview/${reviewId}`);
        return response.data;
    }
}

export default ReviewService;
